lib/core/utils/get-selector.js

```diff
@@ -1,38 +1,166 @@
 const escapeSelector = axe.utils.escapeSelector;
 let isXHTML;
 
-function isUncommonClassName (className) {
-  return ![
-    'focus', 'hover',
-    'hidden', 'visible',
-    'dirty', 'touched', 'valid', 'disable',
-    'enable', 'active', 'col-'
-  ].find(str => className.includes(str));
+/**
+ * get the attribute name and value as a string
+ * @param {Attribute} at    The attribute
+ * @return {String}
+ */
+function getAttributeNameValue(node, at) {
+  const name = at.name;
+  let atnv;
+  
+  if (name.indexOf('href') !== -1 || name.indexOf('src') !== -1) {
+      let value = encodeURI(axe.utils.getFriendlyUriEnd(node.getAttribute(name)));
+      if (value) {
+        atnv = escapeSelector(at.name) + '$="' + value + '"';  
+      } else {
+        return;
+      }
+  } else {
```

Comment:
For the long attributes: long attributes are also likely to be unique so ignoring them will make the algorithm slower and the resulting selector longer because of having to traverse up the tree. How about we put this into Coconut and see what the results look like across multiple sites?

Reply:
Even so, there has to be a cut-off point IMO. Attributes can include entire scripts, hundreds of chars long. In your previous PR, running it on different sites I saw all sorts of stuff, like hashes and serialised JSON in attributes. Those may be unique, but there's a good chance that there are also others, just slightly different, on that same page.

What I'd want from these selectors is that, as far as possible, a dev should be able to read them and know what element we're talking about, without having to go search the DOM for it. I think the longer a selector is, the more likely it is that it's some piece of dynamically injected data, instead of something that was hand coded into the page.